drf_spectacular/contrib/rest_auth.py

```diff
@@ -0,0 +1,154 @@
+from django.conf import settings as global_settings
+
+from rest_framework import serializers
+
+from allauth.account import app_settings as allauth_settings
+
+from rest_auth.serializers import JWTSerializer, TokenSerializer
+
+from drf_spectacular.extensions import OpenApiViewExtension
+from drf_spectacular.utils import extend_schema
+
+SignupSerializer = global_settings.get('REST_AUTH_REGISTER_SERIALIZERS', {}).get('REGISTER_SERIALIZER', None)
+UserSerializer = global_settings.get('REST_AUTH_SERIALIZERS', {}).get('USER_DETAILS_SERIALIZER', None)
+
+if getattr(global_settings, 'REST_USE_JWT', False):
+    AuthTokenSerializer = JWTSerializer
+else:
+    AuthTokenSerializer = TokenSerializer
+
+
+class DRFDefaultDetailSerializer(serializers.Serializer):
+    detail = serializers.CharField(read_only=True, required=False)
```

Comment:
This is one of the responses for https://github.com/tfranzel/drf-spectacular/issues/101

Reply:
Note that is would be highly desirable to fix #101 first!  This is a major bug in the current schemas emitted by drf-spectacular.

Doing that first will cement in the serializer name, and thus the component name, which will be in the schema, and then it can be used there and elsewhere.

Reply:
@tfranzel , please comment here.  This is the most important part, and you ignored it.

Reply:
i have not ignored it. my time is also limited and i wanted to give a well thought out review. will comment on it very soon.